core/storage/classifier/gae_models.py

```diff
@@ -95,19 +100,19 @@ def create(
                 data used by the classifier.
 
         Returns:
-            ID of the new ClassifierModel entry.
+            ID of the new ClassifierDataModel entry.
 
         Raises:
             Exception: A model with the same ID already exists.
         """
 
         instance_id = cls._generate_id(exp_id)
```

Comment:
We don't generate instance_id in ClassifierDataModel at all. We use job_request_id directly as instance_id of ClassifierDataModel. If we use job_request_id as instance_id then there is no need to generate instance_id at all. The random hash generator function will be used in TrainClassifierJobModel. So there we will generate new IDs, here we will use job_request_id directly. Does that clarify your doubt?